frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_language_view.dart

```diff
@@ -12,85 +14,86 @@ class SettingsLanguageView extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return SingleChildScrollView(
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.start,
-        children: [
-          Row(
-            children: [
-              FlowyText.medium(LocaleKeys.settings_menu_language.tr()),
-              const LanguageSelectorDropdown(),
-            ],
-          ),
-        ],
+      child: BlocBuilder<AppearanceSettingsCubit, AppearanceSettingsState>(
+        builder: (context, state) => Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Row(
+              children: [
+                Expanded(
+                  child:
+                      FlowyText.medium(LocaleKeys.settings_menu_language.tr()),
+                ),
+                LanguageSelector(currentLocale: state.locale),
+              ],
+            ),
+          ],
+        ),
       ),
     );
   }
 }
 
-class LanguageSelectorDropdown extends StatefulWidget {
-  const LanguageSelectorDropdown({
-    Key? key,
-  }) : super(key: key);
+class LanguageSelector extends StatelessWidget {
+  final Locale currentLocale;
+  const LanguageSelector({
+    super.key,
+    required this.currentLocale,
+  });
 
   @override
-  State<LanguageSelectorDropdown> createState() =>
-      _LanguageSelectorDropdownState();
-}
+  Widget build(BuildContext context) {
+    return AppFlowyPopover(
+      direction: PopoverDirection.bottomWithRightAligned,
+      child: FlowyTextButton(
+        languageFromLocale(currentLocale),
+        fontColor: Theme.of(context).colorScheme.onBackground,
+        fillColor: Colors.transparent,
+        onPressed: () {},
```

Comment:
unused?

Reply:
As long as the child is clicked, it will show the widget from `popupBuilder`, so `FlowyTextButton` actually just used for showing the appearance here. 
You can refer the `Popover` code here, which is inside the `AppFlowyPopover`
https://github.com/AppFlowy-IO/AppFlowy/blob/49caf74ee19968d73241340bd262714e383bf140/frontend/appflowy_flutter/packages/appflowy_popover/lib/src/popover.dart#L195-L199